### ISSUE TITLE
Test failures on master with Python 3.9

### ISSUE THREAD

**Issue by thesamesam** [byllyfish/precis_i18n#9]
Hi,

I found that running https://github.com/byllyfish/precis_i18n/commit/1498def50914b8d4f670874cc282072790619399 solved test failures from 1.0.1 (with Python 3.8), but I've hit the following with Python 3.9.0b5:

```
python3.9 setup.py test --verbose
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
writing precis_i18n.egg-info/PKG-INFO
writing dependency_links to precis_i18n.egg-info/dependency_links.txt
writing top-level names to precis_i18n.egg-info/top_level.txt
reading manifest file 'precis_i18n.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'precis_i18n.egg-info/SOURCES.txt'
running build_ext
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
writing precis_i18n.egg-info/PKG-INFO
writing dependency_links to precis_i18n.egg-info/dependency_links.txt
writing top-level names to precis_i18n.egg-info/top_level.txt
reading manifest file 'precis_i18n.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'precis_i18n.egg-info/SOURCES.txt'
running build_ext
test_decode (test_codec.TestCodec) ... ok
test_encode (test_codec.TestCodec) ... ok
test_encode_errors (test_codec.TestCodec) ... ok
test_search_function (test_codec.TestCodec) ... ok
test_coalesce (test_codepointset.TestCodepointSet) ... ok
test_contains (test_codepointset.TestCodepointSet) ... ok
test_equals (test_codepointset.TestCodepointSet) ... ok
test_even_odd (test_codepointset.TestCodepointSet) ... ok
test_len (test_codepointset.TestCodepointSet) ... ok
test_malformed_range (test_codepointset.TestCodepointSet) ... ok
test_parse (test_codepointset.TestCodepointSet) ... ok
test_repr (test_codepointset.TestCodepointSet) ... ok
test_derived_props (test_derived_props.TestDerivedProperties)
Compare derived properties against a "golden" file using built-in ... ERROR
test_derived_props_unicodedata2 (test_derived_props.TestDerivedProperties)
Compare derived properties against a "golden" file using ... skipped 'unicodedata2 not available'
test_derived_props (test_derived_props_files.TestDerivedPropsFiles)
Check derived property values do not change as UCD version increases. ... ok
test_iana_derived_props (test_derived_props_files.TestDerivedPropsFiles)
Compare IANA precis-tables to derived-props-6.3.txt ... ok
test_missing (test_factory.TestGetProfile) ... ok
test_unicodedata_arg (test_factory.TestGetProfile) ... ok
test_enforce (test_factory.TestNickname) ... ok
test_enforce (test_factory.TestNicknameCaseMapped) ... ok
test_constructor (test_factory.TestUsername) ... ok
test_constructor_fail (test_factory.TestUsername) ... ok
test_enforce (test_factory.TestUsernameCaseMapped) ... ok
test_enforce (test_factory.TestUsernameCasePreserved) ... ok
test_identifier_oddities (test_factory.TestUsernameCasePreserved) ... ok
test_invalid_argument (test_factory.TestUsernameCasePreserved) ... ok
test_golden_json (test_golden.TestGolden) ... ERROR
test_all_codepoints (test_idempotent.IdempotentTestCase)
Verify all individual code points are idempotent. ... ok
test_broken_profile (test_idempotent.IdempotentTestCase)
Test that we can catch a profile that is not idempotent. ... ok
test_bidi_rule_ltr (test_precis.TestBidiRule) ... ok
test_bidi_rule_rtl (test_precis.TestBidiRule) ... ok
test_has_rtl (test_precis.TestBidiRule) ... ok
test_derived_property (test_precis.TestDerivedProperty) ... ok
test_arabic_indic (test_precis.TestPrecisContextRule) ... ok
test_context_rule (test_precis.TestPrecisContextRule) ... ok
test_extended_arabic_indic (test_precis.TestPrecisContextRule) ... ok
test_katatana_middle_dot (test_precis.TestPrecisContextRule) ... ok
test_rule_greek_keraia (test_precis.TestPrecisContextRule) ... ok
test_rule_hebrew_punctuation (test_precis.TestPrecisContextRule) ... ok
test_rule_middle_dot (test_precis.TestPrecisContextRule) ... ok
test_rule_zero_width_joiner (test_precis.TestPrecisContextRule) ... ok
test_rule_zero_width_nonjoiner (test_precis.TestPrecisContextRule) ... ok
test_invalid_freeform (test_precis.TestPrecisFreeformClass) ... ok
test_valid_freeform (test_precis.TestPrecisFreeformClass) ... ok
test_invalid_identifier (test_precis.TestPrecisIdentifierClass) ... ok
test_valid_identifier (test_precis.TestPrecisIdentifierClass) ... ok
test_arabic_indic (test_precis.TestPrecisUnicodeData) ... ok
test_combining_virama (test_precis.TestPrecisUnicodeData) ... ok
test_control (test_precis.TestPrecisUnicodeData) ... ok
test_default_ignorable_code_point (test_precis.TestPrecisUnicodeData) ... ok
test_extended_arabic_indic (test_precis.TestPrecisUnicodeData) ... ok
test_greek_script (test_precis.TestPrecisUnicodeData) ... ok
test_has_compat (test_precis.TestPrecisUnicodeData) ... ok
test_hebrew_script (test_precis.TestPrecisUnicodeData) ... ok
test_hiragana_katakana_han_script (test_precis.TestPrecisUnicodeData) ... ok
test_noncharacter (test_precis.TestPrecisUnicodeData) ... ok
test_old_hangul_jamo (test_precis.TestPrecisUnicodeData) ... ok
test_replace_whitespace (test_precis.TestPrecisUnicodeData) ... ok
test_valid_join_type (test_precis.TestPrecisUnicodeData) ... ok
test_version_to_float (test_precis.TestPrecisUnicodeData) ... ok
test_width_map (test_precis.TestPrecisUnicodeData) ... ok

======================================================================
ERROR: test_derived_props (test_derived_props.TestDerivedProperties)
Compare derived properties against a "golden" file using built-in
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/tmp/portage/dev-python/precis-i18n-1.0.1_p20190922/work/precis_i18n-1498def50914b8d4f670874cc282072790619399/test/test_derived_props.py", line 20, in test_derived_props
    self._test_derived_props(UnicodeData())
  File "/var/tmp/portage/dev-python/precis-i18n-1.0.1_p20190922/work/precis_i18n-1498def50914b8d4f670874cc282072790619399/test/test_derived_props.py", line 39, in _test_derived_props
    with open(goldpath) as golden:
FileNotFoundError: [Errno 2] No such file or directory: '/var/tmp/portage/dev-python/precis-i18n-1.0.1_p20190922/work/precis_i18n-1498def50914b8d4f670874cc282072790619399/test/derived-props-13.0.txt'

======================================================================
ERROR: test_golden_json (test_golden.TestGolden)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/tmp/portage/dev-python/precis-i18n-1.0.1_p20190922/work/precis_i18n-1498def50914b8d4f670874cc282072790619399/test/test_golden.py", line 28, in test_golden_json
    self.check_disallow(profile, input_, error)
  File "/var/tmp/portage/dev-python/precis-i18n-1.0.1_p20190922/work/precis_i18n-1498def50914b8d4f670874cc282072790619399/test/test_golden.py", line 50, in check_disallow
    input_.encode(profile)
LookupError: unknown encoding: UsernameCaseMapped:ToLower

----------------------------------------------------------------------
Ran 61 tests in 28.989s

FAILED (errors=2, skipped=1)
Test failed: <unittest.runner.TextTestResult run=61 errors=2 failures=0>
error: Test failed: <unittest.runner.TextTestResult run=61 errors=2 failures=0>
```

**Comment by byllyfish** [comment]
Thanks for the report!

For the first test failure, we need to add a 'derived-props-13.0.txt' file for Unicode 13.0 used in Python 3.9. You can run the test_derived_props.py main function using Python 3.9b5 to generate the file. i.e.

```
python3.9 test/test_derived_props.py > test/derived-props-13.0.txt
```

The other test failure is a codec lookup error I haven't seen before. I wonder if there is some underlying Python3.9 change/optimization? (case sensitivity or it doesn't like the colon?). I won't have time to install Python 3.9b5 until later this week.

**Comment by byllyfish** [comment]
Fixed in 1.0.2
